### ISSUE TITLE
[SciPy]: tokenizer error: Expecting terminating symbol for subroutine

### ISSUE THREAD

**Issue by Smit-create** [lfortran/lfortran#672]
Error:
```
% lfortran --fixed-form --implicit-typing --show-asr scipy/optimize/__nnls/nnls.f
tokenizer error: Expecting terminating symbol for subroutine
  --> scipy/optimize/__nnls/nnls.f:90:7
   |
90 |    30 CONTINUE  
   |       ^ 
```

**Comment by Smit-create** [comment]
Similar error:
```
 % lfortran --fixed-form --implicit-typing --show-asr scipy/optimize/cobyla/trstlp.f
tokenizer error: Expecting terminating symbol for subroutine
   --> scipy/optimize/cobyla/trstlp.f:487:7
    |
487 |   500 CONTINUE
    |       ^ 
```

**Comment by Smit-create** [comment]
```
% lfortran --fixed-form --implicit-typing --show-asr scipy/optimize/minpack2/dcsrch.f
tokenizer error: Expecting terminating symbol for subroutine
   --> scipy/optimize/minpack2/dcsrch.f:325:7
    |
325 |    10 continue
    |       ^ 


```

**Comment by konradha** [comment]
I'll enable `CONTINUE` after #679 has been merged.

**Comment by Pranavchiku** [comment]
```
tokenizer error: Expecting terminating symbol for subroutine
   --> /home/pranavchiku/scipy/scipy/sparse/linalg/_eigen/arpack/ARPACK/SRC/dsapps.f:138:7
    |
138 |       include   'debug.h'
    |       ^ 
```

**Comment by konradha** [comment]
> ```
> 
> tokenizer error: Expecting terminating symbol for subroutine
> 
>    --> /home/pranavchiku/scipy/scipy/sparse/linalg/_eigen/arpack/ARPACK/SRC/dsapps.f:138:7
> 
>     |
> 
> 138 |       include   'debug.h'
> 
>     |       ^ 
> 
> ```

#605

**Comment by Pranavchiku** [comment]
```
tokenizer error: Expecting terminating symbol for subroutine
   --> /home/pranavchiku/scipy/scipy/special/cdflib/cdft.f:190:7
    |
190 |       ELSE IF ((3).EQ. (which)) THEN
    |       ^ 

```

**Comment by Pranavchiku** [comment]
```
tokenizer error: Expecting terminating symbol for function
  --> /home/pranavchiku/scipy/scipy/special/mach/i1mach.f:29:7
   |
29 |       EQUIVALENCE (IMACH(4),OUTPUT), (RMACH,SMALL(1))
   |       ^ 
```

**Comment by konradha** [comment]
@Pranavchiku If you could write a reproducer for the `ELSE IF` thing I'll fix it right away. For the `EQUIVALENCE` error there is another issue #692

**Comment by Pranavchiku** [comment]
Doing it

**Comment by Pranavchiku** [comment]
```
      SUBROUTINE cdft(which,p,q,t,df,status,bound)
  170 IF (.NOT. (which.EQ.1)) qporq = p .LE. q
      ELSE IF ((2).EQ. (which)) THEN
      ELSE IF ((3).EQ. (which)) THEN
  310 END IF
      RETURN
      END
```
This produces the same error @konradha

**Comment by konradha** [comment]
Your example does not compile with gfortran and I'm trying to figure out what's the issue for this involved nesting.

**Comment by konradha** [comment]
Also @Pranavchiku @Smit-create -- `tokenizer error: Expecting terminating symbol for subroutine` is not a single issue class, please see if you can find out what's the underlying issue we have to tackle. Thanks :)

**Comment by konradha** [comment]
Apart from the example containing "include", these parse just fine.
